stack_data/core.py

```diff
@@ -421,6 +424,8 @@ def variables_in_executing_piece(self):
             for var, node in self.variables_by_lineno[lineno]
         )
 
+    def __repr__(self):
+        return f"<{self.__class__.__name__} of file {repr(self.filename)} at 0x{id(self)}>"
```

Comment:
Above you used `__qualname__`, here you used `__name__`. It doesn't really matter though.

`0x{id(self)}` needs `:x`.

If we're using format specifiers, `!r` might be nicer than `repr`, but that doesn't really matter.

I think a useful `__repr__` for this should contain `self.frame`, in which case it doesn't need self.filename.